libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -150,9 +159,18 @@ void AC_WPNav::wp_and_spline_init(float speed_cms, Vector3f stopping_point)
     
     // sanity check parameters
     // check _wp_accel_cmss is reasonable
-    _scurve_accel_corner = angle_to_accel(_pos_control.get_lean_angle_max_cd() * 0.01) * 100;
-    const float wp_accel_cmss = MIN(_wp_accel_cmss, _scurve_accel_corner);
-    _wp_accel_cmss.set_and_save_ifchanged((_wp_accel_cmss <= 0) ? WPNAV_ACCELERATION : wp_accel_cmss);
+    const float accel_corner = angle_to_accel(_pos_control.get_lean_angle_max_cd() * 0.01) * 100.0;
+    const float wp_accel_cmss = MIN(_wp_accel_cmss, accel_corner);
+    _wp_accel_cmss.set_and_save_ifchanged((!is_positive(wp_accel_cmss)) ? WPNAV_ACCELERATION : wp_accel_cmss);
```

Comment:
lets remove the set_and_save_ifchanged(), and just use the calculated value, not change the user param
particularly as we now have a tuning knob for the max lean angle, so we could drive the user param to close to zero (1 cd)

Reply:
I agree this set_and_save() is a real bug introduced by PR https://github.com/ArduPilot/ardupilot/pull/17673.  Sorry I did not notice the fact that the position controller lean angle max parameter could drive down the maximum acceleration.

Once fixed we should backport to 4.3.